modules/n1ql/pages/n1ql-language-reference/objectfun.adoc

```diff
@@ -540,36 +540,63 @@ SELECT OBJECT_NAMES(special_flights[*]) AS names;
 ====
 
 [[fn-obj-pairs,OBJECT_PAIRS()]]
-== OBJECT_PAIRS(`expression`)
+== OBJECT_PAIRS(`expression` [, `options` ])
 
-_Alias_: *OBJECT_OUTER_PAIRS(`expression`)*
+_Alias_: *OBJECT_OUTER_PAIRS(`expression` [, `options` ])*
 
 === Description
 
-This function returns an array of objects, containing the names and values of each attribute in the input object.
-It is particularly useful when iterating over multiple objects in an array, as it collates the values from similarly-named attributes into a single nested array.
+This function returns an array of objects, containing the names and values of each attribute in the input object. 
+It also provides an option to return attribute types instead of values. 
+
+The function is particularly useful when iterating over multiple objects in an array, as it collates the values from similarly-named attributes into a single nested array. 
+However, if an object does not contain a shared attribute name, it returns a null entry, similar to an OUTER JOIN.
 
-In this case, the function returns a null entry from any object which does not contain the shared attribute name, rather like an OUTER JOIN.
 For an illustration, refer to the examples below.
 
 === Arguments
 
 expression:: An expression representing an object.
 
+options:: [Optional]
+A JSON object specifying options for the function.
+
+
+=== Options
+
+[options="header", cols="1a,3a,1a"]
+|===
+| Name | Description | Schema
+
+| **types** +
+__required__
+| Determines whether to return attribute types or values.
+
+If TRUE, the function returns the name and type of each attribute.
+
+If FALSE, the function returns the name and value of each attribute.
+
+*Default:* `FALSE`
+| Boolean
+|===
+
 === Return Value
 
-An array of objects, each containing two attributes:
+An array of objects, each containing the following attributes:
 
 name:: The name of an attribute in the source object.
```

Comment:
It's up to you, but for consistency I think we should probably reformat the return value to use a table with exactly the same layout as the options table.